index.js

```diff
@@ -0,0 +1,80 @@
+'use strict';
+const imagemin = require('imagemin');
+const imageminMozjpeg = require('imagemin-mozjpeg');
+const imageminPngquant = require('imagemin-pngquant');
+const Svgo = require('svgo');
+const fs = require('fs');
+const os = require('os');
+const async = require('async');
+const path = require('path');
+
+const compress = (file, level, callback) => {
+  const handleResult = result => {
+    if (result.length === 0) {
+      return callback('Unable to compress image');
+    }
+    callback(null, result[0].path);
+  };
+  const options = {
+    plugins: [imageminMozjpeg({ quality: level }), imageminPngquant({ quality: level })]
+  };
+  imagemin([file], os.tmpdir(), options).then(handleResult);
+};
+
+const svgOptimize = (file, callback) => {
+  fs.readFile(file, 'utf8', (err, data) => {
+    if (err) {
+      return callback(err);
+    }
+    new Svgo().optimize(data, (result) => {
+      fs.writeFile(file, result.data, callback);
+    });
+  });
+};
+
+const crop = (im, pathToFile, origin, dims, gravity, callback) => {
+  const gm = require('gm').subClass({ imageMagick: im });
+  const pathToOutput = pathToFile;
+  gm(pathToFile)
+  .gravity(gravity)
+  .crop(dims[0], dims[1], origin[0], origin[1])
+  .write(pathToOutput, (err) => {
+    callback(err, pathToOutput);
+  });
+};
+
+module.exports.process = (imageFilePath, options, callback) => {
```

Comment:
do we need a separate function?  couldn't it just be `require('optimiz')` and then `optimize(file, options)`